subcommand/delete-completed-job/command.go

```diff
@@ -0,0 +1,170 @@
+package deletecompletedjob
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/hashicorp/consul-k8s/subcommand"
+	k8sflags "github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/hashicorp/go-hclog"
+	"github.com/mitchellh/cli"
+	v1 "k8s.io/api/batch/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"os"
+	"sync"
+	"time"
+)
+
+// Command is the command for deleting completed jobs.
+type Command struct {
+	UI cli.Ui
+
+	flags         *flag.FlagSet
+	k8s           *k8sflags.K8SFlags
+	flagNamespace string
+	flagTimeout   string
+
+	once      sync.Once
+	help      string
+	k8sClient kubernetes.Interface
+
+	// retryDuration is how often we'll retry deletion.
+	retryDuration time.Duration
+}
+
+func (c *Command) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+
+	c.k8s = &k8sflags.K8SFlags{}
+	c.flags.StringVar(&c.flagNamespace, "k8s-namespace", "",
+		"Name of Kubernetes namespace where the job is deployed")
+	c.flags.StringVar(&c.flagTimeout, "timeout", "30m",
+		"How long we'll wait for the job to complete before timing out, e.g. 1ms, 2s, 3m")
+	flags.Merge(c.flags, c.k8s.Flags())
+	c.help = flags.Usage(help, c.flags)
+
+	// Default retry to 30s. This is exposed for setting in tests.
+	if c.retryDuration == 0 {
+		c.retryDuration = 30 * time.Second
```

Comment:
I realized this should be 1s because helm upgrade (and install I think) will wait until the job exits before the helm command exits. So if it waits 30s when the acl-init completes in 2s then it's just sitting unnecessarily.